app/scenes/channel_members/channel_members.js

```diff
@@ -3,44 +3,134 @@
 
 import React, {PropTypes, PureComponent} from 'react';
 import {
+    Alert,
     InteractionManager,
     StyleSheet,
     View
 } from 'react-native';
+import {injectIntl, intlShape} from 'react-intl';
 
 import MemberList from 'app/components/member_list';
 
+import ChannelMembersTitle from './channel_members_title';
+import RemoveMemberButton from './remove_member_button';
+
 const style = StyleSheet.create({
     container: {
         flex: 1
     }
 });
 
-export default class ChannelMembers extends PureComponent {
+class ChannelMembers extends PureComponent {
     static propTypes = {
+        intl: intlShape.isRequired,
         currentChannel: PropTypes.object,
         currentChannelMembers: PropTypes.array.isRequired,
         currentChannelMemberCount: PropTypes.number.isRequired,
         currentTeam: PropTypes.object,
         preferences: PropTypes.object,
         requestStatus: PropTypes.string,
+        isAdmin: PropTypes.bool.isRequired,
+        subscribeToHeaderEvent: React.PropTypes.func,
+        unsubscribeFromHeaderEvent: React.PropTypes.func,
         actions: PropTypes.shape({
-            getProfilesInChannel: PropTypes.func.isRequired
+            getProfilesInChannel: PropTypes.func.isRequired,
+            goBack: PropTypes.func.isRequired,
+            handleRemoveChannelMembers: PropTypes.func.isRequired
         })
     }
 
+    static navigationProps = {
+        renderTitleComponent: () => {
+            return <ChannelMembersTitle/>;
+        },
+        renderRightComponent: (props, emitter) => {
+            return <RemoveMemberButton emitter={emitter}/>;
+        }
+    }
+
+    state = {
+        selectedMembers: {}
+    }
+
+    componentWillMount() {
+        this.props.subscribeToHeaderEvent('remove_members', this.handleRemoveMembersPress);
+    }
+
     componentDidMount() {
         InteractionManager.runAfterInteractions(() => {
             this.props.actions.getProfilesInChannel(this.props.currentTeam.id, this.props.currentChannel.id, 0);
         });
     }
 
+    componentWillUnmount() {
+        this.props.unsubscribeFromHeaderEvent('remove_members');
+    }
+
+    handleRemoveMembersPress = () => {
+        const {selectedMembers} = this.state;
+        const membersToRemove = Object.keys(selectedMembers).filter((m) => selectedMembers[m]);
+
+        const {formatMessage} = this.props.intl;
+        if (!membersToRemove.length) {
+            Alert.alert(
+                formatMessage({
+                    id: 'mobile.routes.channel_members.action',
+                    defaultMessage: 'Remove Members'
+                }, {
+                    term: 'Remove'
```

Comment:
I'm trying to make these reusable for the add members screen as well because I think it needs an alert. ```term``` is for add/remove and ```prep``` is for to/from. That alert still needs to be added, which I guess begs the question, do we want an alert on that other screen? 

If so I can do it in another PR or just add it to this one.